core/src/main/scala/kafka/server/BrokerToControllerChannelManager.scala

```diff
@@ -171,6 +184,20 @@ class BrokerToControllerRequestThread(
   private val requestQueue = new LinkedBlockingDeque[BrokerToControllerQueueItem]()
   private var activeController: Option[Node] = None
 
+  private val lock = new ReentrantReadWriteLock
+
+  def activeControllerAddress(): Option[Node] = {
+    CoreUtils.inReadLock(lock) {
```

Comment:
This might be overkill. Perhaps we could make `activeController` volatile or an atomic reference?